flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterEntrypoint.java

```diff
@@ -183,6 +184,9 @@ public void startCluster() throws ClusterEntrypointException {
 
             SecurityContext securityContext = installSecurityContext(configuration);
 
+            Thread.setDefaultUncaughtExceptionHandler(
+                    new ClusterUncaughtExceptionHandler(
+                            configuration.get(ClusterOptions.UNCAUGHT_EXCEPTION_HANDLING)));
```

Comment:
Maybe move this logic into the `ClusterEntrypointUtils.configureUncaughtExceptionHandler(Configuration)` and then reuse it for the `TaskManagerRunners`.